server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -594,4 +604,18 @@ public static Set<String> getPossibleRoleNames() {
         return roleMap.keySet();
     }
 
+    @Override
+    public Principal getPrincipal() {
+        return this.principal;
+    }
```

Comment:
I put the getPrincipal method at this level since I was not sure whether there may future classes that extend the DiscoveryNode class. Currently, DiscoveryNode is only used for ExtensionDiscoveryNode so it does not make a ton of difference but I assumed it was set up this way because there may be other types of discovery nodes in the future. 

I will drop the subject interface for now and open a follow up PR later if we need to move this to the Extension specific level.